tools/cbmc/proofs/CBMCStubLibrary/portableMemoryAllocation.c

```diff
@@ -0,0 +1,17 @@
+#include "FreeRTOS.h"
+
+/* xWantedSize is not bounded in this function,
+but there might be a need to bound it in the future.
+In theory, CBMC malloc allows to allocate an
+arbitray ammount of data.*/
+void *pvPortMalloc( size_t xWantedSize ){
+	if(xWantedSize == 0){
+		return NULL;
+	}
+	uint8_t byte;
+	return byte ? malloc(xWantedSize) : NULL;
```

Comment:
Shouldn't you use a version of `malloc` that considers it could fail (i.e., return `NULL`)? I agree with your comment about the need to bound `xWantedSize`, see our version of [bounded_malloc](https://github.com/awslabs/aws-c-common/blob/48995ff6da2a4547430e74e3415b28346956166c/.cbmc-batch/source/proof_allocators.c#L69). Perhaps you could included it in this code base (another PR), and then use it here.

Reply:
`pvPortMalloc` is used in the codebase instead of direct calls to malloc. It is up to the third-party code to provide an implementation. Pure malloc is not used in the code base. Therefore, I think as long as `pvPortMalloc` allows to fail, this is a sufficient good model for the FreeRTOS code base.